### PR TITLE
fcosKola: Automatically pick up tests/kola 

### PR DESCRIPTION
**Pull request by cgwalters**
So we can use `-E` in CI, see
https://github.com/coreos/coreos-assembler/pull/1252

### CODE REVIEW

**Comment by cgwalters** [comment]
(Totally didn't test this)

**Comment by jlebon** [comment]
Ahh yup, you're right we need to enhance this step here for https://github.com/coreos/fedora-coreos-config/pull/305.

Hmm, how about instead of this, we automatically pass `--exttest src/config/tests/kola` if we find that the directory exists? In general, I want to establish conventions to make it dead easy to do the right thing.

**Comment by jlebon** [comment]
I can pick this up if you'd like!

**Comment by cgwalters** [comment]
>  we automatically pass --exttest src/config/tests/kola

Oh definitely we should do that yep.  But this approach is needed for the ostree case...unless we define conventions for that.

**Comment by jlebon** [comment]
Yeah, I think that makes sense. I.e. let's define `tests/kola` as the external tests dir. For config dirs like fedora-coreos-config, we check for this via `${cosaDir}/src/config/tests/kola`. For the project being built, we just check `tests/kola` (i.e. the workspace). This matches e.g. `make: true` blindly building the code in the workspace.

For things that break that convention or for tests stored elsewhere, we can add an `exttests: []` arg that override the default list? But that can come later.

**Comment by cgwalters** [comment]
Oh I like that yep!

**Comment by cgwalters** [comment]
https://github.com/coreos/coreos-assembler/pull/1261

**Comment by cgwalters** [comment]
And reworked this per above discussion.

File: vars/fcosKola.groovy [hunk 1]
```diff
@@ -3,8 +3,13 @@ def call(cosaDir = "/srv/fcos") {
     stage('Kola') {
         parallel run: {
             stage("run") {
+                def args = ""
+                def srctests = new File("tests/kola").getAbsolutePath()
+                if (srctests.isDirectory()) {
```
Comment (jlebon) [inline vars/fcosKola.groovy]: This unfortunately doesn't work in Kubernetes pods, e.g. the pipeline has: https://github.com/coreos/fedora-coreos-pipeline/blob/2c6c49599fd242e3ffbc6d4b4b1087fdb731a5ae/utils.groovy#L46. Which, yeah OK let me upstream the `shwrap_rc` function to this repo: https://github.com/coreos/coreos-ci-lib/pull/17. Then we can use it here for this.

I'll double check whether this is fixed, and if not, also upstream `path_exists` as a helper.

Comment (cgwalters) [reply vars/fcosKola.groovy]: Ah, nice catch.  OK I took another stab at this but still not tested.

**Comment by jlebon** [comment]
I just did a minor fixup to this to avoid another round-trip. (Changed `WORKSPACE` to `env.WORKSPACE` and just spelled out the `--exttest` switch to be more explicit.)

Thanks!

**Comment by cgwalters** [comment]
I pushed https://github.com/coreos/coreos-ci-lib/commit/55bb2ae8e693beea2249dfc37e8e6968c0b978c5

**Pull request merged**